plugins/ccgx/fu-ccgx-pure-hid-device.c

```diff
@@ -0,0 +1,538 @@
+/*
+ * Copyright (C) 2023 Framework Computer Inc
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-ccgx-common.h"
+#include "fu-ccgx-firmware.h"
+#include "fu-ccgx-hpi-common.h"
+#include "fu-ccgx-pure-hid-device.h"
+
+#define ROW_SIZE 128
+#define FW1_START 0x0030
+#define FW2_START 0x0200
+#define FW1_METADATA 0x03FF
+#define FW2_METADATA 0x03FE
+#define WRITE_HEADER_SIZE 4
+
+/**
+ * FU_CCGX_PURE_HID_DEVICE_IS_IN_RESTART:
+ *
+ * Device is in restart and should not be closed manually.
+ *
+ * Since: 1.7.0
+ */
+#define FU_CCGX_PURE_HID_DEVICE_IS_IN_RESTART (1 << 0)
+
+struct _FuCcgxPureHidDevice {
+	FuHidDevice parent_instance;
+	FuCcgxFwMode operating_mode;
+	guint32 versions[FU_CCGX_FW_MODE_LAST];
+	//guint8 bootloader_info;
+	//guint8 bootmode_reason;
+	guint32 silicon_id;
+	//guint32 image_1_row;
+	//guint32 image_2_row;
+	//device_uid: // [u8; 6],
```

Comment:
why are these commented out?  Trying to decide whether to make them properties of the device or not?

Reply:
Yep. Probably won't keep them. Maybe device_uid is worth putting here? Will these feed into the GUID?

Reply:
Can you use uid to better differentiate what firmware goes with what device? If so then you can probably parse the header and match it to ensure no one ever flashes a DP firmware to the HDMI for example.

Reply:
No, UID is different for each device. We can avoid flashing the wrong firmware just by looking at the PID.
I mean...once you flash the wrong one, the PID will change. But that shouldn't happen, I think

Reply:
I've removed then now I've shifted to rustgen.